lewis/core/statemachine.py

```diff
@@ -294,6 +295,8 @@ def bind_handlers_by_name(self, instance, override=False, prefix=None):
         """
         if prefix is None:
             prefix = {}
+        if not isinstance(prefix, dict) and hasattr(prefix, '__iter__'):
```

Comment:
A `set` should work fine from what I can tell. It's just an unordered, unique-only list really. The only unexpected behaviour would arise from the unordered nature of it... but that's just you getting exactly what you asked for by using a `set`. :smile: 

Further, `isinstance(OrderedDict(), dict)` returns `True`, so there shouldn't be any issues with those kinds of collections either.